challenge_1/c/xoumi/ReverseString.c

```diff
@@ -0,0 +1,27 @@
+#include<stdio.h>
+void main()
+{
+    int i,j;
+    char *string,temp;
+    printf("Enter string: ");
+    //because scanf is scary
+    gets(string);
+
+    //Just a loop to count length instead of using strlen because this isn't scary
+    for(i=0;string[i]!='\0';i++);
```

Comment:
While you can write your own strlen for learning purposes you really shouldn't. Strlen can be optimized by the compiler to be much more efficient. Just take a look at the glibc implementation http://www.stdlib.net/~colmmacc/strlen.c.html